godot-cell/tests/mock_blocking.rs

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! A mock implementation of our instance-binding pattern in pure rust for the blocking variant of GdCell.
+//!
+//! Used so we can run miri on this, which we cannot when we are running in itest against Godot.
+
+use std::collections::HashMap;
+use std::error::Error;
+use std::marker::PhantomData;
+use std::sync::{atomic::AtomicUsize, Mutex, OnceLock};
+
+use godot_cell::blocking::{GdCell, InaccessibleGuard};
+
+struct InstanceBinding(*mut ());
+
+unsafe impl Sync for InstanceBinding {}
+unsafe impl Send for InstanceBinding {}
+
+static INSTANCE_BINDINGS: OnceLock<Mutex<HashMap<usize, InstanceBinding>>> = OnceLock::new();
+
+struct InstanceStorage<T> {
+    cell: GdCell<T>,
+}
+
+fn binding() -> &'static Mutex<HashMap<usize, InstanceBinding>> {
+    INSTANCE_BINDINGS.get_or_init(|| Mutex::new(HashMap::new()))
+}
+
+fn register_instance<T>(instance: T) -> usize {
+    static COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+    let key = COUNTER.fetch_add(1, std::sync::atomic::Ordering::AcqRel);
+
+    let binding = binding();
+
+    let mut guard = binding.lock().unwrap();
+
+    assert!(!guard.contains_key(&key));
+
+    let cell = GdCell::new(instance);
+    let storage = Box::new(InstanceStorage { cell });
+    let ptr = Box::into_raw(storage) as *mut ();
+
+    guard.insert(key, InstanceBinding(ptr));
+    key
+}
+/*
+unsafe fn free_instance<T>(key: usize) {
+    let binding = binding();
+    let mut guard = binding.lock().unwrap();
+
+    let InstanceBinding(ptr) = guard.remove(&key).unwrap();
+
+    let ptr: *mut InstanceStorage<T> = ptr as *mut _;
+
+    let storage = unsafe { Box::from_raw(ptr) };
+}
+*/
```

Comment:
Do you plan to still implement this, or is there another reason it's commented out?